lua/ttt2/libraries/orm.lua

```diff
@@ -0,0 +1,261 @@
+	local where = {}
+	local primaryKey = self._primaryKey
+	local primaryKeyCount = #primaryKey
+
+	if not istable(primaryValue) and primaryKeyCount == 1 then
+		where = sql.SQLIdent(primaryKey[1]) .. "=" .. sql.SQLStr(primaryValue)
+	elseif istable(primaryValue) and #primaryValue == primaryKeyCount then
+		for i = 1, primaryKeyCount do
+			where[i] = sql.SQLIdent(primaryKey[i]) .. "=" .. sql.SQLStr(primaryValue[i])
+		end
+
+		where = table.concat(where, " AND ")
+	else
+		ErrorNoHalt("[ORM] Number of primaryvalues does not match number of primarykeys!")
+
+		return false
+	end
+
+	local result = sql.QueryRow("SELECT * FROM " .. sql.SQLIdent(self._tableName) .. " WHERE " .. where)
+
+	if result then
+		return self:New(result)
+	end
+end
+
+---
+-- Creates a new object of the model.
+-- @param[opt] table data Preexisting data the object should be initialized with.
+-- @return ORMOBJECT The created object.
+-- @realm shared
+function ORMMODEL:New(data)
+	local object = data or {}
+
+	object._tableName = self._tableName
+	object._dataStructure = self._dataStructure
+
+	-- DO NOT setup delete/save/refresh functions if no primarykey is found.
+	-- In those cases the 'rowid' column would function as the primarykey, but as the rowid could change anytime (https://www.sqlite.org/rowidtable.html) data could be deleted unintentionally.
+	-- Most likely rowids wont change in gmod as there is no vacuum operation but just to be safe we will not allow to use such tables. ref: https://wiki.facepunch.com/gmod/sql
+
+	if not self._primaryKey then
+		return object
+	end
+
+	object.Delete = ORMOBJECT.Delete
+	object.Save = ORMOBJECT.Save
+	object.Refresh = ORMOBJECT.Refresh
+	object._primaryKey = self._primaryKey
+	object._primaryKeyList = self._primaryKeyList
+	object._columnList = self._columnList
+
+	return object
+end
+
+---
+-- Retrieves all saved objects of the model with the given filters from the database.
+-- @param table filters An array of filters. Each filter should contain a `column`, `op`, `value` and `concat`(if it is not the last filter).
+-- @return table Returns an array of all found objects.
+-- @realm shared
+function ORMMODEL:Where(filters)
+	local query = "SELECT * FROM " .. sql.SQLIdent(self._tableName) .. " WHERE "
+	local whereList = {}
+
+	for i = 1, #filters do
+		local curFilter = filters[i]
+
+		whereList[i] = sql.SQLIdent(curFilter[column]) .. (curFilter[op] or "=") .. sql.SQLStr(curFilter[value]) .. (curFilter[concat] or "")
+	end
+
+	whereList = table.concat(whereList)
+
+	query = query .. whereList
+
+	local objects = sql.Query(query)
+
+	if not self._primaryKey then
+		return objects
+	end
+
+	for i = 1, #objects do
+		objects[i].Save = ORMOBJECT.Save
+		objects[i].Delete = ORMOBJECT.Delete
+		objects[i].Refresh = ORMOBJECT.Refresh
+	end
+
+	return objects
+end
+
+---
+-- @class ORMOBJECT
+
+---
+-- Deletes the given object from the database storage.
+-- @return nil|boolean Returns false if an error occurred, nil otherwise.
```

Comment:
That's definitely a bad behavior for code style, because `if not ORMOBJECT:Delete() then` will ever be run (if returning just `nil` or `false`.